cmd/skopeo/sync.go

```diff
@@ -647,6 +651,31 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) (retErr error) {
 
 	for _, srcRepo := range srcRepoList {
 		options.SourceCtx = srcRepo.Context
+		// only proceed for top refs if defined
+		if opts.topRefs > 0 {
+			// Check Refs follow Semantic Versioning
+			semverRefs, nonSemverRefs := filterOutRefsNonSemver(srcRepo.ImageRefs)
+			if len(nonSemverRefs) > 0 {
+				logrus.Warnf("Filtered out %d ref(s) are not follow Semver in total %d ref(s)", len(nonSemverRefs), len(srcRepo.ImageRefs))
+			}
+			// Sort semverRefs
+			sort.Slice(semverRefs, func(i, j int) bool {
+				v1, _ := version.NewVersion(semverRefs[i].DockerReference().(reference.Tagged).Tag())
+				v2, _ := version.NewVersion(semverRefs[j].DockerReference().(reference.Tagged).Tag())
+				return v1.LessThan(v2)
+			})
+			topRefs := opts.topRefs
+			totalRefs := len(srcRepo.ImageRefs)
+			totalSemverRefs := len(semverRefs)
+			if totalSemverRefs == 0 {
+				return errors.New("ERROR: Not found any refs follow Semantic versioning. Check your tag's regex configuration OR doesn't it follow Semantic versioning?")
+			}
+			if topRefs > totalSemverRefs {
+				topRefs = totalSemverRefs
+			}
+			logrus.Infof("Only sync top largest %d ref(s) follow Semver in %d ref(s) from the total %d ref(s)", topRefs, totalSemverRefs, totalRefs)
```

Comment:
All of these messages would really work _much_ better if they indicated which registry/repo (or whatever the scope of the filter is) they are talking about. There can be dozens of entries in `srcRepoList`, and without including the relevant scope these messages would just be numbers with no indication what they relate to.

So I suppose `repoDescriptor` needs to have some kind of `sourceDescription string` field added… Or maybe this code needs to be involved before a `repoDescriptor` is even created, with the source description provided by the caller in context.